go16 1.6.3

dylanplecki opened the pull request:
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-versions/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
- [X] Does your submission adhere to Versions "Acceptable Formulae" [guidelines](https://github.com/Homebrew/homebrew-versions/blob/07b17dfb9f1/README.md#acceptable-formulae)?

-----

Added golang version 1.6.3 as `go16` formula due to recent update to 1.7 in core repository.

MikeMcQuaid commented on go16.rb:
```diff
@@ -0,0 +1,128 @@
+  mirror "https://fossies.org/linux/misc/go1.6.3.src.tar.gz"
+  version "1.6.3"
+  sha256 "6326aeed5f86cf18f16d6dc831405614f855e2d416a91fd3fdc334f772345b00"
+  go_version = "1.6"
```
Can inline this variable.

MikeMcQuaid commented on go16.rb:
```diff
@@ -0,0 +1,128 @@
+    sha256 "54159189e4779b8c34235bd3f18c62122b4826f478a0a6c9812fbcce608849bf" => :el_capitan
+    sha256 "597524370e994f7d153e6ae20ed28a4ad9fee1ea9e2d8a7b29674699a52ae601" => :yosemite
+    sha256 "41a1322a0c302b9d7c74788f7d57cffc1296b627e77165507106412b3932d44a" => :mavericks
+  end
```
Remove this bottle block.

DomT4 commented:
This will need a `depends_on MaximumMacOSRequirement => :el_capitan` as well, since this won't build on Sierra.

dylanplecki commented:
Added the maximum MacOS requirement, removed the bottle block, and inlined the Go tools version number with the latest commit. Thanks for the suggestions!

MikeMcQuaid commented:
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for new formulae is `go16 1.6.3 (new formula)`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md). Thanks!
